packages/odyssey-react-mui/src/labs/DataView/DataView.tsx

```diff
@@ -210,10 +211,18 @@ const DataView = <TData extends MRT_RowData>({
     ],
   );
 
-  const getRowId = useCallback<Required<DataViewProps<TData>>["getRowId"]>(
-    (originalRow) => originalRow.id as string,
-    [],
-  );
+  const getRowId = useCallback(
+    (row: TData, index: number, parentRow: MRT_Row<TData>) => {
+      if (getRowIdProp) {
+        return getRowIdProp(row, index, parentRow);
+      }
+      if (row.id) {
+        return row.id as string;
+      }
+      return createUniqueId();
+    },
+    [getRowIdProp],
+  ) as (row: TData, index: number, parentRow?: MRT_Row<TData>) => string;
```

Comment:
The `as` and the function look the same. We shouldn't need those here right? We should know the `id` is a string at this point I'd guess.